### PR TITLE
Add manual Otel instrumentation for Spans

### PR DESCRIPTION
**Pull request by sjuarezgx**
Issue:107378

### CODE REVIEW

File: gxobservability/pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            <version>1.23.0-alpha</version>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-extension-annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
+            <version>1.23.0-alpha</version>
```
Comment (ggallotti) [inline gxobservability/pom.xml:45]: Can we bump to version 1.36?

File: gxobservability/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        </dependency>
+    </dependencies>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.opentelemetry</groupId>
+                <artifactId>opentelemetry-bom</artifactId>
+                <version>1.23.0</version>
```
Comment (ggallotti) [inline gxobservability/pom.xml:53]: Can we bump to version 1.36?

File: gxobservability/pom.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	<build>
+		<finalName>gxobservability</finalName>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
```
Comment (ggallotti) [inline gxobservability/pom.xml:68]: Why target JDK8?

**Review by ggallotti — COMMENTED** [review]
Left dome minor comments

File: gxobservability/pom.xml (hunk 1 above)
Comment (sjuarezgx) [reply gxobservability/pom.xml]: Done

File: gxobservability/pom.xml (hunk 2 above)
Comment (sjuarezgx) [reply gxobservability/pom.xml:53]: Bumped to the latest.

File: gxobservability/pom.xml (hunk 3 above)
Comment (sjuarezgx) [reply gxobservability/pom.xml]: Removed the JDK version.

**Review by ggallotti — APPROVED** [review]

**Pull request merged**